trigger/src/lib.rs

```diff
@@ -0,0 +1,290 @@
+
+#[global_allocator]
+static ALLOC: GlobalDlmalloc = GlobalDlmalloc;
+
+#[derive(Deserialize)]
+struct Config {
+    /// Operation mode of this trigger - hub or domestic.
+    ///
+    /// Alters the behaviour of how transfers are made upon verification.
+    mode: OperationMode,
+    /// Storage with admin-only write access. Contains [`ChainSnapshot`]\(s).
+    ///
+    /// The trigger is deployed separately for each connected chain. Thus, there is
+    /// only one trigger on each domestic chain (domestic-hub), and many triggers
+    /// on the hub chain (hub-\*domestic). Therefore, on the hub chain, we use
+    /// a single admin storage to store multiple chain snapshots. We deploy a separate
+    /// trigger for each chain working with its own entry in the admin store.
+    admin_store: NftId,
+    /// Points to the exact key in the admin storage containing [`ChainSnapshot`]
+    /// for this trigger to work with.
+    admin_store_chain_key: Name,
+    /// Storage to which the relay has write access. Contains [`RelayBlockMessage`].
+    relay_store: NftId,
+    /// Points to the exact key in the relay storage containing [`RelayBlockMessage`]
+    /// this trigger will read.
+    relay_store_message_key: Name,
+    /// Global information about all the chains in the hub chain network.
+    chains: BTreeMap<ChainId, ChainConfig>,
+}
+
+/// Trigger operation mode
+#[derive(Deserialize)]
+#[serde(tag = "type")]
+enum OperationMode {
+    /// Trigger is deployed on the hub chain
+    Hub,
+    /// Trigger is deployed on a domestic chain
+    Domestic(ChainId),
+}
+
+/// Information about a specific chain
+#[derive(Deserialize)]
+struct ChainConfig {
+    /// Omnibus account of that chain presented on the chain _this trigger is deployed on_.
+    omnibus_account: AccountId,
+}
+
+/// Message provided by an untrusted relay that the trigger is going to verify
+#[derive(Deserialize)]
+struct RelayBlockMessage {
+    /// Header of the new block. Must be consequent to the block in [`ChainSnapshot`].
+    header: BlockHeader,
+    /// Block signatures. Will be verified against validators set in [`ChainSnapshot`].
+    signatures: BTreeSet<SignatureOf<BlockHeader>>,
+    /// _Interesting_ transactions in the given block.
+    ///
+    /// Interesting means they contain e.g. [`Transfer`] instructions that the trigger will
+    /// scan and act on. Currently, it is only transfers; later, it could include
+    /// [`RegisterBox::Peer`] (and unregister) to change the validators set.
+    ///
+    /// The goal of the PoC is to show that a relay does not need to send the **entire block**
+    /// (i.e. all of its transactions), but only a subset of them. Trigger can verify them using
+    /// their [`CommittedTransaction::entrypoint_proof`] against [`BlockHeader::merkle_root`].
+    interesting_transactions: Vec<CommittedTransaction>,
+}
+
+/// Memory about a chain.
+#[derive(Deserialize, Serialize)]
+struct ChainSnapshot {
```

Comment:
There may be a conventional term for this.
Candidates might include `LightClientState`, `CheckPoint`, or `TrustAnchor`.